src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultPluginOptions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.KeyVault
+{
+    /// <summary>
+    /// The endpoint options used to create a new <see cref="KeyVaultPlugin"/>.
+    /// </summary>
+    public class KeyVaultPluginOptions
```

Comment:
Do not allow write properties. Instead, have a constructor that would validate options when the object is constructed.